pkg/backup/resource_backupper.go

```diff
@@ -237,43 +239,86 @@ func (rb *defaultResourceBackupper) backupResource(group *metav1.APIResourceList
 
 		log.Infof("Retrieved %d items", len(items))
 
+		// back up CRD for resource if found. We should only need to do this is IncludeClusterResources is nil. If it's false
+		// we don't want to back it up, and if it's true it will already be included.
+		if len(items) > 0 && rb.backupRequest.Spec.IncludeClusterResources == nil {
```

Comment:
We'll end up doing this once per namespace, if specific namespaces are specified. It would be a little more efficient to do this just once, after L255 (i.e. after the range over `namespacesToList`), assuming we backed up >0 items.

Reply:
Sure. This makes sense -- combined with the below comment, this would just mean keeping track of either a count of items backed up or a boolean "at least one item backed up" across namespaces, then outside the namespace loop we can make the CRD backup call.